src/Symfony/Component/Lock/Store/PdoStore.php

```diff
@@ -332,34 +332,17 @@ private function getDriver(): string
         } else {
             $driver = $con->getDriver();
 
-            switch (true) {
-                case $driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver:
-                    throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
-                case $driver instanceof \Doctrine\DBAL\Driver\AbstractMySQLDriver:
-                    $this->driver = 'mysql';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOSqlite\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\SQLite\Driver:
-                    $this->driver = 'sqlite';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOPgSql\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\PgSQL\Driver:
-                    $this->driver = 'pgsql';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\OCI8\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOOracle\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\OCI\Driver:
-                    $this->driver = 'oci';
-                    break;
-                case $driver instanceof \Doctrine\DBAL\Driver\SQLSrv\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDOSqlsrv\Driver:
-                case $driver instanceof \Doctrine\DBAL\Driver\PDO\SQLSrv\Driver:
-                    $this->driver = 'sqlsrv';
-                    break;
-                default:
-                    $this->driver = \get_class($driver);
-                    break;
+            if ($driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver) {
+                throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
             }
+
+            $this->driver = [
+                'mssql' => 'sqlsrv',
+                'oracle' => 'oci',
+                'postgresql' => 'pgsql',
+                'sqlite' => 'sqlite',
+                'mysql' => 'mysql',
+            ][$driver->getDatabasePlatform()->getName()] ?? \get_class($driver);
```

Comment:
The `getName()` method has been deprecated recently. We should work with `instanceof` instead.

Reply:
It has been deprecated in 3.2 (which is unreleased IIRC); we support 2.x and 3.x, are we sure that we can rely on that extensively?

Reply:
Doing `$driver->getDatabasePlatform() instanceof MysqlPlatform` (or similar checks) works in both 2.x and 3.x

Reply:
We can leave it as is for now, if you prefer. It's just that we will need to change this piece of code again sooner or later.